// gondolajs/webpack.telemetry.js
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

//@ts-check

'use strict';
var nodeExternals = require('webpack-node-externals');
const path = require('path');


/**@type {import('webpack').Configuration}*/
const config = {
    output: {
        filename: 'telemetry.js',
        library: "gondola-telemetry",       // Important
        libraryTarget: 'umd',               // Important
        umdNamedDefine: true                // Important
    },
    target: 'node',
    mode: 'production',
    node: {
        __filename: true,
        __dirname: true
    },

    externals: [nodeExternals({
        whitelist: [
            'gondola-telemetry'
        ]
    })]
}

module.exports = config;
